### PR TITLE
Usage and command line arguments

### LINKED ISSUE #191: cIRI usage
Write a complete usage for cIRI, based on arguments

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes #191 

# Test Plan:
Not applicable

### CODE REVIEW

File: ciri/conf/conf.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  memset(conf, 0, sizeof(ciri_conf_t));
+  conf->api_port = CONF_DEFAULT_API_PORT;
+  conf->neighbors = CONF_DEFAULT_NEIGHBORS;
+  conf->conf_file = CONF_DEFAULT_CONF_FILE;
+  conf->tcp_receiver_port = CONF_DEFAULT_TCP_RECEIVER_PORT;
+  conf->udp_receiver_port = CONF_DEFAULT_UDP_RECEIVER_PORT;
+  conf->testnet = CONF_DEFAULT_TESTNET;
+  conf->remote = CONF_DEFAULT_REMOTE;
+  conf->remote_auth_token = CONF_DEFAULT_REMOTE_AUTH_TOKEN;
+  conf->remote_limite_api = CONF_DEFAULT_REMOTE_LIMIT_API;
```
Comment (tsvisabo) [inline ciri/conf/conf.c]: typo

File: ciri/main.c [hunk 2]
```diff
@@ -27,17 +26,14 @@ int main(int argc, char* argv[]) {
...
+  log_info(MAIN_LOGGER_ID, "Initializing configuration\n");
   if (ciri_conf_init(&core_g.config)) {
-    log_critical(MAIN_LOGGER_ID,
-                 "Initializing configuration variables failed\n");
+    log_critical(MAIN_LOGGER_ID, "Initializing configuration failed\n");
     return EXIT_FAILURE;
   }
 
-  log_info(MAIN_LOGGER_ID, "Parsing configuration variables\n");
+  log_info(MAIN_LOGGER_ID, "Parsing command lien arguments\n");
```
Comment (tsvisabo) [inline ciri/main.c]: typo

File: ciri/conf/conf_provider_getopt.c [hunk 3]
```diff
@@ -24,7 +24,7 @@ retcode_t ciri_conf_parse(ciri_conf_t* conf, int argc, char** argv) {
         conf->udp_receiver_port = atoi(optarg);
         break;
       default:
-        break;
+        return RC_CONF_INVALID_ARGUMENTS;
```
Comment (tsvisabo) [inline ciri/conf/conf_provider_getopt.c]: RC_CIRI_CONF_INVALID_ARGUMENTS

File: ciri/conf/conf_args.h [hunk 4]
```diff
@@ -8,6 +8,53 @@
...
+  NO_ARG,
+  REQUIRED_ARG,
+  OPTIONAL_ARG
+} ciri_arg_requirement_t;
+
+static struct ciri_argument_s {
+  char* long_option;
+  char short_option;
+  char* desc;
+  ciri_arg_requirement_t status;
```
Comment (tsvisabo) [inline ciri/conf/conf_args.h]: should be requirement rather than status i think

File: ciri/conf/usage.h [hunk 5]
```diff
@@ -8,57 +8,64 @@
...
      "Exclude certain API calls from being able to be accessed remotely",
      REQUIRED_ARG},
-    {"--send-limit", 's',
+    {"send-limit", 's',
      "Limit the outbound bandwidth consumption. Limit is set to mbit/s",
      REQUIRED_ARG},
-    {"--max-peers", 'm',
-     " Limit the number of max accepted peers. Default is set to 0 (mutual "
+    {"max-peers", 'm',
+     "Limit the number of max accepted peers. Default is set to 0 (mutual "
```
Comment (tsvisabo) [inline ciri/conf/usage.h]: Limits

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: ciri/main.c (hunk 2 above)
Comment (thibault-martinez) [reply ciri/main.c]: You're reviewing an old commit, that's fixed

File: ciri/conf/conf_args.h (hunk 4 above)
Comment (thibault-martinez) [reply ciri/conf/conf_args.h]: Old commit, it's now `has_arg` to match getop struct field `has_arg`

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**